tests/rbd_mirror/test_rbd_clone_mirror.py

```diff
@@ -0,0 +1,104 @@
+    4. Enable pool mood journal based mirroring on the pool respectively.
+    5. Verify for the clone image is also get mirrored.
+    6. Create one more clone from the image and verify it is mirrored.
+    """
+
+    try:
+        log.info("Running RBD mirroring test case for clone images")
+        config = kw.get("config")
+        rbd = Rbd(**kw)
+
+        poolname = rbd.random_string() + "9521pool"
+        imagename = rbd.random_string() + "9521image"
+        snap = rbd.random_string()
+        clone1 = rbd.random_string()
+        size = "10G"
+
+        # create pool and image on cluster1
+        size = kw.get("size", "10G")
+        if not rbd.create_pool(poolname=poolname):
+            log.error(f"Pool creation failed for pool {poolname}")
+            return 1
+        rbd.create_image(pool_name=poolname, image_name=imagename, size=size)
+
+        # create clone of the image
+        rbd.snap_create(poolname, imagename, snap)
+        snap_name = f"{poolname}/{imagename}@{snap}"
+        cmd = "ceph osd set-require-min-compat-client mimic"
+        rbd.exec_cmd(cmd=cmd)
+        rbd.create_clone(snap_name, poolname, clone1)
+
+        # enable journal based mirroring for pool mode
+        mirror1, mirror2 = [
+            rbdmirror.RbdMirror(cluster, config)
+            for cluster in kw.get("ceph_cluster_dict").values()
+        ]
+        imagespec = poolname + "/" + imagename
+        mirror2.create_pool(poolname=poolname)
+        mirror1.config_mirror(mirror2, poolname=poolname, mode="pool")
+        mirror1.enable_mirror_image(poolname, imagename, "journal")
+        mirror2.wait_for_status(poolname=poolname, images_pattern=1)
+        mirror1.wait_for_status(imagespec=imagespec, state_pattern="up+stopped")
+        mirror2.wait_for_status(imagespec=imagespec, state_pattern="up+replaying")
+
+        # verify for cloned image mirrored
+        imagespec1 = poolname + "/" + clone1
+        mirror1.enable_mirror_image(poolname=poolname, imagename=clone1, mode="journal")
+        mirror2.wait_for_status(poolname=poolname, images_pattern=2)
+        mirror1.wait_for_status(imagespec=imagespec1, state_pattern="up+stopped")
+        mirror2.wait_for_status(imagespec=imagespec1, state_pattern="up+replaying")
+
+        # Create clone of the image and verify for the image mirrored status
+        clone2 = rbd.random_string()
+        imagespec2 = poolname + "/" + clone2
+        rbd.create_clone(snap_name, poolname, clone2)
+        mirror2.wait_for_status(poolname=poolname, images_pattern=3)
+        mirror1.wait_for_status(imagespec=imagespec2, state_pattern="up+stopped")
+        mirror2.wait_for_status(imagespec=imagespec2, state_pattern="up+replaying")
+
+        # clean up the configuration
+        mirror1.delete_image(imagespec)
+        mirror1.delete_image(imagespec1)
+        mirror1.delete_image(imagespec2)
+        mirror1.clean_up(peercluster=mirror2, pools=[poolname])
```

Comment:
May be we should strengthen clean_up to required steps actual cleanup

Reply:
@VasishtaShastry  can you elaborate more on this.

Reply:
Will create a new task in next sprint to add changes in clean_up steps.

Reply:
Ref https://github.com/red-hat-storage/cephci/issues/2054